src/backend/distributed/connection/connection_management.c

```diff
@@ -766,3 +772,80 @@ AfterXactHostConnectionHandling(ConnectionHashEntry *entry, bool isCommit)
 		}
 	}
 }
+
+
+/*
+ * SetCitusNoticeProcessor sets the NoticeProcessor to DefaultCitusNoticeProcessor
+ */
+void
+SetCitusNoticeProcessor(MultiConnection *connection)
+{
+	PQsetNoticeProcessor(connection->pgConn, DefaultCitusNoticeProcessor,
+						 connection);
+}
+
+
+/*
+ * SetCitusNoticeLevel is used to set the notice level for distributed
+ * queries.
+ */
+void
+SetCitusNoticeLevel(int level)
+{
+	CitusNoticeLogLevel = level;
+}
+
+
+/*
+ * UnsetCitusNoticeLevel sets the CitusNoticeLogLevel back to
+ * its default value.
+ */
+void
+UnsetCitusNoticeLevel()
+{
+	CitusNoticeLogLevel = DEFAULT_CITUS_NOTICE_LEVEL;
+}
+
+
+/*
+ * DefaultCitusNoticeProcessor is used to redirect worker notices
+ * from logfile to console.
+ */
+static void
+DefaultCitusNoticeProcessor(void *arg, const char *message)
+{
+	MultiConnection *connection = (MultiConnection *) arg;
+	char *nodeName = connection->hostname;
+	uint32 nodePort = connection->port;
+	char *chompedMessage = pchomp(message);
+	char *trimmedMessage = TrimLogLevel(chompedMessage);
+
+	ereport(CitusNoticeLogLevel, (errmsg("%s", trimmedMessage),
+								  errdetail("while executing command on %s:%d\n",
```

Comment:
there should not be `\n` at the end here.

One minor nitpicky thing: The "while executing command" sounds a bit weird in the middle of all the output. The phrasing is based on getting errors, so you'd see "ERROR...while executing command". Maybe we could say "message from %s:%d" or something like that. 

(if you want to go all crazy, you could include the original log level here, e.g. "info from localhost:9700")